src/frontends/lean/definition_cmds.cpp

```diff
@@ -863,12 +863,15 @@ environment single_definition_cmd_core(parser & p, decl_cmd_kind kind, cmd_meta
     } catch (throwable & ex) {
         // Even though we catch exceptions during elaboration, there can still be other exceptions,
         // e.g. when adding a declaration to the environment.
-        try {
-            auto res = process(p.mk_sorry(header_pos, true));
+
+        // prefer elaboration error messages
+        if (!lt.get().has_entry_now(is_error_message))
             p.mk_message(header_pos, ERROR).set_exception(ex).report();
```

Comment:
I'm not really sure why this change is needed.

Reply:
I'll improve the comment. The change hides kernel exceptions after error recovery, which do nothing but make the original error harder to see.